init.php

```diff
@@ -2,66 +2,89 @@
 
 require_once __DIR__ . '/vendor/autoload.php';
 
-use Symfony\Component\Console\Helper\Table;
 use Symfony\Component\Console\Command\Command;
-use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Console\Helper\Table;
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\BufferedOutput;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\SingleCommandApplication;
+use Symfony\Component\Console\Style\SymfonyStyle;
 
-CONST VALIDATOR_URL = 'https://validator.prestashop.com';
+const VALIDATOR_URL = 'https://validator.prestashop.com';
 
 (new SingleCommandApplication())
     ->setName('ValidatorDecider') // Optional
     ->setVersion('1.0.0') // Optional
     ->addArgument('github_link', InputArgument::OPTIONAL, 'The github_link')
     ->addArgument('github_branch', InputArgument::OPTIONAL, 'The github_branch')
+    ->addArgument('archive', InputArgument::OPTIONAL, 'The archive')
     ->setCode(function (InputInterface $input, OutputInterface $output) {
 
         $buffer = new BufferedOutput($output->getVerbosity());
         $io = new SymfonyStyle($input, $buffer);
         $client = new \GuzzleHttp\Client([
             'base_uri' => VALIDATOR_URL
         ]);
-        
+
+        $apiKey = getenv('VALIDATOR_API_KEY');
+
+        if (empty($apiKey)) {
+            throw new Exception('No API Key is set to authenticate the request to the validator. Please set the env var VALIDATOR_API_KEY');
+        }
+
+        $multipart = [
+            [
+                'name'     => 'key',
+                'contents' => $apiKey
+            ]
+        ];
+
         // Call validator
         try {
+            if ($input->hasArgument('archive')) {
+                $archive = $input->getArgument('archive');
+
+                if (empty($archive) || !file_exists($archive) || !is_readable($archive)) {
+                    throw new Exception(sprintf('File %s was not found, or cannot be read', $archive));
+                }
+
+                $multipart[] = [
+                    'name'     => 'archive',
+                    'contents' => fopen($archive, 'r'),
+                ];
+            } else {
+                $multipart[] = [
+                    'name'     => 'github_link',
+                    'contents' => $input->getArgument('github_link'),
+                ];
+                $multipart[] = [
+                    'name'     => 'github_branch',
+                    'contents' => $input->getArgument('github_branch'),
+                ];
+            }
```

Comment:
I also meant moving this part outside the try{} block.
The reason is you throw an exception that I would prefer seeing uncaught, so we can display the whole exception content instead of seeing only the message.